include/orc/parse_file.hpp

```diff
@@ -31,17 +31,22 @@ struct freader {
     // `<=` here because sometimes we jump to one past the end of the buffer right before stopping.
     explicit operator bool() const { return static_cast<bool>(_buffer) && _p <= _l; }
 
-    std::size_t size() const { return _l - _p; }
+    std::size_t size() const {
+        assert(*this);
```

Comment:
Checking: did you mean `*this` or `this` or `this && *this`

Reply:
I do mean `*this`. Because it is in an `assert` clause, that will require the compiler convert the expression to a boolean value. `freader` has an `explicit operator bool` that allows `freader` to convert to a `boolean` operation, and does so by performing a couple of checks to make sure the internal state of the instance is valid. So by writing `assert(*this)`, we are asserting some necessary validity checks for pre/postconditions on a handful of the `freader` APIs.